freeenergyframework/stats.py

```diff
@@ -69,15 +73,27 @@ def unique_differences(x):
         N = len(x)
         return np.array( [ (x[i] - x[j]) for i in range(N) for j in range(N) if (i != j) ] )
 
+    if dy_true is None:
+        dy_true = np.zeros_like(y_true)
+    if dy_pred is None:
+        dy_pred = np.zeros_like(y_pred)
     assert len(y_true) == len(y_pred)
```

Comment:
this could be

`assert len(y_true) == len(y_pred) == len(dy_true) == len(dy_pred)`